src/codegeneration/InnerForOnTransformer.js

```diff
@@ -0,0 +1,230 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import alphaRenameThisAndArguments from './alphaRenameThisAndArguments.js';
+import {
+  parseStatement,
+  parseStatements
+} from './PlaceholderParser.js';
+import {
+  AnonBlock,
+  Block,
+  BreakStatement,
+  ContinueStatement,
+  LabelledStatement,
+  ReturnStatement
+} from '../syntax/trees/ParseTrees.js';
+import {
+  createArgumentList,
+  createAssignmentStatement,
+  createAssignmentExpression,
+  createCaseClause,
+  createDefaultClause,
+  createIdentifierExpression as id,
+  createNumberLiteral,
+  createSwitchStatement,
+  createThisExpression,
+  createVariableDeclaration,
+  createVariableDeclarationList,
+  createVariableStatement,
+  createVoid0
+} from './ParseTreeFactory.js';
+import {ARGUMENTS} from '../syntax/PredefinedName.js';
+import {VAR} from '../syntax/TokenType.js';
+import {VARIABLE_DECLARATION_LIST} from '../syntax/trees/ParseTreeType.js';
+
+export class InnerForOnTransformer extends ParseTreeTransformer {
+
+  constructor(idGenerator, labelSet) {
+    this.idGenerator_ = idGenerator;
+    this.inLoop_ = 0;
+    this.inBreakble_ = 0;
+    this.variableDeclarations_ = [];
+    this.extractedStatements_ = [];
+    this.labelSet_ = labelSet;
+    this.labelledStatements_ = {};
```

Comment:
Funny, I corrected it in my editor but the edit must have been lost.

2015-02-06 18:44 GMT+01:00 Erik Arvidsson <notifications@github.com>:

> In src/codegeneration/InnerForOnTransformer.js
> <https://github.com/google/traceur-compiler/pull/1684#discussion_r24257436>
> :
>
> > +  createVoid0
> > +} from './ParseTreeFactory.js';
> > +import {ARGUMENTS} from '../syntax/PredefinedName.js';
> > +import {VAR} from '../syntax/TokenType.js';
> > +import {VARIABLE_DECLARATION_LIST} from '../syntax/trees/ParseTreeType.js';
> > +
> > +export class InnerForOnTransformer extends ParseTreeTransformer {
> > +
> > +  constructor(idGenerator, labelSet) {
> > +    this.idGenerator_ = idGenerator;
> > +    this.inLoop_ = 0;
> > +    this.inBreakble_ = 0;
> > +    this.variableDeclarations_ = [];
> > +    this.extractedStatements_ = [];
> > +    this.labelSet_ = labelSet;
> > +    this.labelledStatements_ = {};
>
> This was the one that thought you had not updated the code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/traceur-compiler/pull/1684/files#r24257436>.
>